### ISSUE TITLE
OpenTracing support

### ISSUE THREAD

**Issue by npepinpe** [zeebe-io/zeebe#2932]
**Is your feature request related to a problem? Please describe.**

I'd like to have an easy way to trace/breakdown the processing of different components of our system.

**Describe the solution you'd like**

As a start, to measure latency and pinpoint which parts are slower, we can integrate OpenTracing (e.g. Jaeger, Zipkin, or whatever) into our application to trace a single user request across all our moving parts and break it down into several spans, e.g. `Client -> Gateway -> BrokerCommandAPI -> LogStorageAppender -> DistributedLogStreamService -> StreamProcessor -> ServerTransport -> Client`.

This is useful both internally (for us to pinpoint latency hotspots) and can easily be plugged in externally to user's applications if they already use OpenTracing (but I would not do this kind of pluggable implementation stuff in the first iteration).

**Describe alternatives you've considered**

Our metrics are already quite useful, but are aggregates and do not allow us to track down on a per-request basis.

**Additional context**

- https://opentracing.io/
- https://www.jaegertracing.io/
- https://skywalking.apache.org/

**Comment by Zelldon** [comment]
Might be interesting https://lightstep.com/

**Comment by npepinpe** [comment]
I think LightStep implements the OpenTracing API - https://opentracing.io/docs/supported-tracers/

**Comment by npepinpe** [comment]
I investigated a bit more, and I think it shouldn't be too hard to integrate this into Zeebe.

I was initially stumped by the different batching operations we do and how to correlate records to spans, but I realized this is only really useful for us, a normal user doesn't need to know the engine internals and is fine with a blackbox (for a single request).

In that sense, from a user point of view, what we need to implement is:

1. Start client span when the client sends a request (request must be tagged with a unique ID)
1. Start server span "gateway" when the gateway receives the request
1. Start client span "broker" when the gateway sends the request to the broker
1. Start server span "broker" when the request is successfully added to the dispatcher
1. Close server span "broker" when the corresponding response is being sent from the broker
1. Close client span "broker" when the corresponding response is being forwarded to the user
1. Close server span "gateway" when the gateway sent the response to the user
1. Close client span when the client has received the response

I would propose we use the OpenConcensus API for this, and start with Jaeger has an implementation since it works with both Java and Go clients.
